File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.cpp
class Solution {
public:
    int findGCD(vector<int>& nums) {
        int smallest = nums[0];
        int largest = nums[0];
        int gcd = 1;
        for(int i=1; i<nums.size(); i++)
        {
            if(nums[i] < smallest)
                smallest = nums[i];
            if(nums[i] > largest)
                largest = nums[i];
        }

        if(smallest == largest)
            return smallest;
        
        for(int i=1; i <= smallest && i<=largest; i++)
        {
            if(smallest % i == 0 && largest % i == 0)
                gcd = i;
        }

        return gcd;
    }
};

File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.py
class Solution:
    def findGCD(self, nums: List[int]) -> int:
        smallest = nums[0]
        largest = nums[0]
        for i in nums:
            if(i < smallest):
                smallest = i
            if(i > largest):
                largest = i

        if(smallest == largest):
            return smallest
        
        while(largest != 0):
            temp = largest
            largest = smallest % largest
            smallest = temp
        
        return smallest